src/Symfony/Component/PropertyAccess/Tests/PropertyAccessorTest.php

```diff
@@ -97,6 +98,18 @@ public function testGetValue($objectOrArray, $path, $value)
         $this->assertSame($value, $this->propertyAccessor->getValue($objectOrArray, $path));
     }
 
+    /**
+     * Test get dynamic value from object is other than \stdClass instance.
+     */
+    public function testGetDynamicValue()
+    {
+        $value = 'dynamicPropertyValue';
+        $path = 'dynamicProperty';
+        $object = new TestClassDynamicProperty($value);
+
+        $this->assertSame($value, $this->propertyAccessor->getValue($object, $path));
+    }
+
```

Comment:
Maybe you can add a test for the exception path

Reply:
The thrown exception is already covered on access to the existent non-public property (in another unit-test). To assert exact Exception was thrown after attempt property by reference I suggest TestCase::expectExceptionMessage().

Reply:
done